packages/ui/src/components/viewport/EditorPreview.tsx

```diff
@@ -246,6 +252,8 @@ export function EditorPreview() {
             <GridIcon />
           </ButtonCheckbox>
           <ColorPicker />
+          {/* Key is set to the state so that the component renders when the state changes */}
+          <Coordinates visibility={coords} viewState={state} viewSize={size} />
```

Comment:
`size.x` and `size.y` should be added to the `ViewportState`.
The state is already exposed via a context which means that you don't need to pass it as a property.
In `Coordinates` you can just do:
```ts
const state = useContext(ViewportContext);
```
To access the viewport state.

Reply:
So I updated the code to include the `size.x` and `size.y` in the ViewportState. Please let me know if I did not implement it the way you intended. I will look at the remaining 3 comments tomorrow as I need to go to bed.